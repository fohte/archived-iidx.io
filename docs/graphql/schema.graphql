enum ClearLamp {
  """
  Assist Clear
  """
  ASSIST

  """
  Easy Clear
  """
  EASY

  """
  EX Hard Clear
  """
  EX_HARD

  """
  Failed
  """
  FAILED

  """
  FullCombo Clear
  """
  FULL_COMBO

  """
  Hard Clear
  """
  HARD

  """
  Clear
  """
  NORMAL
}

"""
Autogenerated return type of CreateUser
"""
type CreateUserPayload {
  user: User!
}

enum Difficulty {
  """
  Another
  """
  ANOTHER

  """
  Hyper
  """
  HYPER

  """
  Normal
  """
  NORMAL
}

enum Grade {
  """
  A
  """
  A

  """
  AA
  """
  AA

  """
  AAA
  """
  AAA

  """
  B
  """
  B

  """
  C
  """
  C

  """
  D
  """
  D

  """
  E
  """
  E

  """
  F
  """
  F
}

type Map {
  bestResult(username: String!): Result
  difficulty: Difficulty!
  id: ID!
  level: Int!
  maxBpm: Int!
  minBpm: Int!
  music: Music!
  numNotes: Int!
  playStyle: PlayStyle!
}

type Music {
  artist: String!
  genre: String!
  id: ID!
  leggendaria: Boolean!
  map(difficulty: Difficulty!, playStyle: PlayStyle!): Map
  maps: [Map!]!
  series: Int!
  subTitle: String!
  textageUid: String!
  title: String!
}

type Mutation {
  createUser(displayName: String, username: String!): CreateUserPayload
  registerResultsWithCSV(csv: String!, playStyle: PlayStyle!): RegisterResultsWithCSVPayload
}

enum PlayStyle {
  """
  DP
  """
  DP

  """
  SP
  """
  SP
}

type Query {
  """
  Find maps.
  """
  maps: [Map!]

  """
  Find a music.
  """
  music(id: ID!): Music

  """
  Find musics.
  """
  musics: [Music!]

  """
  Find a user by name.
  """
  user(name: String!): User

  """
  The currently authenticated user.
  """
  viewer: User
}

"""
Autogenerated return type of RegisterResultsWithCSV
"""
type RegisterResultsWithCSVPayload {
  success: Boolean!
}

type Result {
  clearLamp: ClearLamp!
  grade: Grade!
  id: ID!
  missCount: Int!
  score: Int!
}

type User {
  id: ID!
  name: String!
  profile: UserProfile!
}

type UserProfile {
  displayName: String!
  id: ID!
}