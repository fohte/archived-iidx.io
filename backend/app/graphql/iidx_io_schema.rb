# frozen_string_literal: true

class IIDXIOSchema < GraphQL::Schema
  mutation(Types::MutationType)
  query(Types::QueryType)
end
