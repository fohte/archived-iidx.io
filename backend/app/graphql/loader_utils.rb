# frozen_string_literal: true

module LoaderUtils
  class << self
    def find_by(*args, **kwargs, &block)
      promise = load_promise(*args, **kwargs)

      if block_given?
        promise.then(&block)
      else
        promise.then { |record| record }
      end
    end

    def find_by!(model, column_values, **kwargs)
      promise = load_promise(model, column_values, **kwargs)

      promise.then do |record|
        raise IIDXIO::GraphQL::NotFoundError, "Couldn't find #{model.name} with #{column_values.to_json}" if record.nil?

        if block_given?
          yield(record)
        else
          record
        end
      end
    end

    private

    def load_promise(model, column_values, scope: nil)
      Loaders::RecordLoader
        .for(model, columns: column_values.keys, scope: scope)
        .load(column_values.values)
    end
  end
end
