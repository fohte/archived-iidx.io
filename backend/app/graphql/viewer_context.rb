# frozen_string_literal: true

class ViewerContext < GraphQL::Query::Context
  def viewer
    self[:viewer]
  end

  def firebase_uid
    self[:firebase_uid]
  end
end
