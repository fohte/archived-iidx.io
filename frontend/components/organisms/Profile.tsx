import * as React from 'react'
import { Header } from 'semantic-ui-react'

import { FindUserUser } from '@app/queries'

export type Props = {
  user: FindUserUser
}

const Profile: React.SFC<Props> = ({ user }) => (
  <>
    <Header as="h2">@{user.name}</Header>
  </>
)
export default Profile
