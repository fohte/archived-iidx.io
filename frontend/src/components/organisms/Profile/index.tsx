import * as React from 'react'

import { PlayStyle } from '@app/queries'

export interface Props {
  playStyle: PlayStyle
}

const Profile: React.SFC<Props> = ({ playStyle }) => <>{playStyle}</>

export default Profile
