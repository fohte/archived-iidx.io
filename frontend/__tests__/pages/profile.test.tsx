import { GraphQLError } from 'graphql'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Profile from '@app/pages/profile'

describe('/', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Profile
        user={{ __typename: 'User', name: 'username' }}
        loading={false}
      />,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when loading', () => {
    const component = renderer.create(<Profile loading />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly if there are errors', () => {
    const errors: Readonly<[GraphQLError]> = [
      new GraphQLError(
        'not found',
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        { code: 'NOT_FOUND' },
      ),
    ]

    const component = renderer.create(
      <Profile errors={errors} loading={false} />,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
