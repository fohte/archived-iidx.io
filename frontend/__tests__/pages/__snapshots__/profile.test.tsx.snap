// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ renders correctly 1`] = `
<div
  className="box"
>
  <nav
    className="headerNav"
  >
    <div
      className="container"
    >
      <div
        className="flexContainer"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <div
            className="item"
          >
            <span>
              iidx.io
            </span>
          </div>
        </a>
        <div
          className="item"
        >
          loading
        </div>
      </div>
    </div>
  </nav>
  <div
    className="content"
  >
    <div
      className="container"
    >
      <div
        className="user-profile-header-wrapper"
      >
        <div
          className="user-profile-header"
        >
          <div
            className="user-profile-content"
          >
            <div
              className="user-avatar"
            />
            <h2>
              @
              username
            </h2>
          </div>
        </div>
        <div
          className="tabs-wrapper"
        >
          <ul
            className="tabs"
          >
            <li
              className="tab-item active"
            >
              <a
                href="/@username/sp"
                onClick={[Function]}
              >
                <span
                  className="icon-text"
                >
                  Overview
                </span>
              </a>
            </li>
            <li
              className="tab-item"
            >
              <a
                href="/@username/sp/musics"
                onClick={[Function]}
              >
                <span
                  className="icon-text"
                >
                  Musics
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      SP
    </div>
  </div>
  <footer
    className="box"
  >
    <div
      className="container"
    >
      <div
        className="inner"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-copyright fa-w-16 "
          data-icon="copyright"
          data-prefix="far"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 448c-110.532 0-200-89.451-200-200 0-110.531 89.451-200 200-200 110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200zm107.351-101.064c-9.614 9.712-45.53 41.396-104.065 41.396-82.43 0-140.484-61.425-140.484-141.567 0-79.152 60.275-139.401 139.762-139.401 55.531 0 88.738 26.62 97.593 34.779a11.965 11.965 0 0 1 1.936 15.322l-18.155 28.113c-3.841 5.95-11.966 7.282-17.499 2.921-8.595-6.776-31.814-22.538-61.708-22.538-48.303 0-77.916 35.33-77.916 80.082 0 41.589 26.888 83.692 78.277 83.692 32.657 0 56.843-19.039 65.726-27.225 5.27-4.857 13.596-4.039 17.82 1.738l19.865 27.17a11.947 11.947 0 0 1-1.152 15.518z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        2018
         iidx.io |
         
        test
      </div>
    </div>
  </footer>
</div>
`;

exports[`/ renders correctly if there are errors 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "#fff",
      "color": "#000",
      "display": "flex",
      "flexDirection": "column",
      "fontFamily": "-apple-system, BlinkMacSystemFont, Roboto, \\"Segoe UI\\", \\"Fira Sans\\", Avenir, \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
      "height": "100vh",
      "justifyContent": "center",
      "textAlign": "center",
    }
  }
>
  <div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "body { margin: 0 }",
        }
      }
    />
    <h1
      style={
        Object {
          "borderRight": "1px solid rgba(0, 0, 0,.3)",
          "display": "inline-block",
          "fontSize": "24px",
          "fontWeight": 500,
          "margin": 0,
          "marginRight": "20px",
          "padding": "10px 23px 10px 0",
          "verticalAlign": "top",
        }
      }
    >
      404
    </h1>
    <div
      style={
        Object {
          "display": "inline-block",
          "height": "49px",
          "lineHeight": "49px",
          "textAlign": "left",
          "verticalAlign": "middle",
        }
      }
    >
      <h2
        style={
          Object {
            "fontSize": "14px",
            "fontWeight": "normal",
            "lineHeight": "inherit",
            "margin": 0,
            "padding": 0,
          }
        }
      >
        This page could not be found
        .
      </h2>
    </div>
  </div>
</div>
`;

exports[`/ renders correctly when loading 1`] = `"loading"`;
